virttest/vt_cluster/__init__.py

```
@@ -0,0 +1,284 @@
+        if os.path.isfile(self._filename):
+            self._data = self._load()
+        else:
+            self._data = self._empty_data
+
+    def _save(self):
+        with open(self._filename, "wb") as f:
+            pickle.dump(self._data, f, protocol=0)
+
+    def _load(self):
+        with open(self._filename, "rb") as f:
+            return pickle.load(f)
```

Ruff (0.12.2)

`pickle` and modules that wrap it can be unsafe when used to deserialize untrusted data, possible security issue

(S301)